### ISSUE TITLE
precompile fails with Julia < 1.6

### ISSUE THREAD

**Issue by minrk** [JuliaInterop/ZMQ.jl#236]
The use of `bind('tcp://localhost:*')` fails precompile on zmq < 4.3.5, because binding to hostnames is not supported in zmq 4.3.2, the latest version packaged by ZeroMQ_jll for Julia < 1.6.

Using 127.0.0.1 will _almost_ always work. I think `*` / `0.0.0.0` will always work, but binding to public interfaces may not be desirable.

You could also lookup localhost's ip with `Sockets.getaddrinfo("localhost")`, which is ultimately what the current code is doing, but the getaddrinfo is [in libzmq itself](https://github.com/zeromq/libzmq/blob/c2fae81460d9d39a896da7b3f72484d23a172fa7/src/ip_resolver.cpp#L721C12-L721C23) in the supported versions.

**Comment by minrk** [comment]
Forgot repro:

`docker run --rm julia:1.5`

then:

```julia
using Pkg
Pkg.add("ZMQ")
pkg"precompile"
```

fails with:

```
julia> pkg"precompile"
Precompiling project...
[ Info: Precompiling ZMQ [c2297ded-f4af-51ae-bb23-16f91089e4e1]
ERROR: LoadError: ZMQ.StateError("No such device")
Stacktrace:
 [1] bind(::ZMQ.Socket, ::String) at /root/.julia/packages/ZMQ/s0VKZ/src/socket.jl:91
 [2] macro expansion at /root/.julia/packages/ZMQ/s0VKZ/src/ZMQ.jl:61 [inlined]
 [3] top-level scope at /root/.julia/packages/PrecompileTools/L8A3n/src/workloads.jl:85
 [4] include(::Function, ::Module, ::String) at ./Base.jl:380
 [5] include(::Module, ::String) at ./Base.jl:368
 [6] top-level scope at none:2
 [7] eval at ./boot.jl:347 [inlined]
 [8] eval(::Expr) at ./client.jl:467
 [9] top-level scope at ./none:3
in expression starting at /root/.julia/packages/ZMQ/s0VKZ/src/ZMQ.jl:47
```

**Comment by JamesWrigley** [comment]
Interesting :thinking: I'm not that inclined to keep supporting versions less than 1.6 since it's the LTS, but if this is all we need for compatibility then it's ok. I'll have a look.

**Comment by minrk** [comment]
I've tested that #237 fixes pre-compile on 1.3 and 1.5 in docker.

FWIW, I'd be totally okay making a release with #237 and then immediately dropping support for < 1.6. The only issue right now is that the current version claims to support 1.3 but doesn't actually install on < 1.6.

**Comment by JamesWrigley** [comment]
Yes, makes sense. I'll keep compatibility for now because this is quite an old package and I'm aware that there are people using it on older Julia versions. If compatibility is trickier later I'll bump the compat bounds.

Anyway, thank you very much for the report and the bug fix :heart: I've registered a new version which should be available soon: https://github.com/JuliaRegistries/General/pull/107787
